logstash-core/lib/logstash/api/lib/app/commands/system/plugins_command.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+require "app/command"
+
+class LogStash::Api::PluginsCommand < LogStash::Api::Command
+
+  def run
+    { :total => plugins.count, :plugins => plugins }
+  end
+
+  private
+
+  def plugins
+    @plugins ||= find_plugins_gem_specs.map do |spec|
+      { :name => spec.name, :version => spec.version.to_s }
+    end
+  end
+
+  def find_plugins_gem_specs
+    @specs ||= Gem::Specification.find_all.select{|spec| logstash_plugin_gem_spec?(spec)}
```

Comment:
We can not call the plugin manager from the logstash-core, as the code is organized right now it makes no sense, I mean without a more deep refactoring of this classes you mention (they do really need to put moved and organized cleaner and better for being reusable).